Strange issue when compiling with MSVC 2022

I am trying to compile PR #241 using MSVC 2022 (version 19.32), but get the following C2653 error:
```
[build] C:\Users\srs\src\stdBLAS\include\experimental\__p1673_bits/transposed.hpp(255,57): error C2653: 'original_mapping_type': is not a class or namespace name [C:\Users\srs\src\stdBLAS\build\tests\native\givens.vcxproj]
[build] C:\Users\srs\src\stdBLAS\include\experimental\__p1673_bits/transposed.hpp(253,17): message : This diagnostic occurred in the compiler generated function 'auto std::experimental::__p1673_version_0::linalg::impl::transposed_layout<std::experimental::layout_stride>::mapping(const std::experimental::layout_stride::mapping<Extents> &)' [C:\Users\srs\src\stdBLAS\build\tests\native\givens.vcxproj]
```
When I use clang++ provided with MSVC 2022, there are no such errors.

--- a/include/experimental/__p1673_bits/transposed.hpp
+++ b/include/experimental/__p1673_bits/transposed.hpp
@@ -251,7 +251,12 @@ namespace impl {
     template<class OriginalExtents>
     static auto mapping(const typename layout_stride::template mapping<OriginalExtents>& orig_map) {
       using original_mapping_type = typename layout_stride::template mapping<OriginalExtents>;
-      using extents_type = transpose_extents_t<typename original_mapping_type::extents_type>;
+      // MSVC 2022 doesn't like the following commented-out line of code.
+      // See https://github.com/kokkos/stdBLAS/issues/242#issuecomment-1174738571
+      //
+      //using extents_type = transpose_extents_t<typename original_mapping_type::extents_type>;
+      using original_extents_type = typename original_mapping_type::extents_type;
+      using extents_type = transpose_extents_t<original_extents_type>;
       using return_mapping_type = typename layout_type::template mapping<extents_type>;
       // NOTE (mfh 2022/07/04) Commented-out code relates
       // to the build error reported in my comment here:

